extender/testdata/cnb/ext/samples_curl/0.0.1/bin/build

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# INPUT ARGUMENTS
+platform_dir=$2
+env_dir=${platform_dir}/env
+output_dir=$1
+plan_path=$3
+
+# Create Dockerfile
+cat << EOF > ${output_dir}/Dockerfile
+ARG base_image
+FROM \${base_image}
+
+# TODO: why wasn't this needed before?
+USER root
```

Comment:
I believe this was not needed before because we were executing the instructions on the builder which was running as root.

This raises a good question/concern. If, for run extensions, we apply `Dockerfile` after `Dockerfile` and the last `Dockerfile` leaves the `USER root` as the last instruction...is that ok? That feels wrong as one of the pillars of CNB is that the resulting image should _not_ be root IMO. It gets a bit muddier when you think about full run image replacements as well. Does the final image need to be using the same `CNB_USER_ID` as the original run image reference to make things happy/correct?

I'm sure @sclevine has some thoughts on this...